client/src/pages/DuelCreate.jsx

```diff
@@ -12,11 +12,35 @@ import PageError from "../components/status/PageError";
 const DuelCreate = ({ history }) => {
   const [searchRegex, setSearchRegex] = useState(null);
   const [selected, setSelected] = useState(null);
+  const [search, setSearch] = useState(null);
   const [error, setError] = useState(null);
   const queryClient = useQueryClient();
 
   const { data: listOfUsers, isSuccess, isError } = useQuery(["users", "challengeable"]);
 
+  const [system, setSystem] = useState(0);
+  const [difficulty, setDifficulty] = useState(1);
+  const [dataSystems, setData] = useState([]);
```

Comment:
Par convention on garde le même nom pour les deux en rajoutant juste `set` pour le setter, et puis dans la mesure du possible éviter le mot *data* qui est souvent trop vague
```suggestion
  const [systemsList, setSystemsList] = useState([]);
```